File: cpp/alerts/unwanted_block.cpp
#include "./unwanted_block.h"
#include <cstdint>

namespace ltrs {

int unwanted_block_alert_get_block_index(lt::unwanted_block_alert* alert) {
    return alert->block_index;
}

PieceIndex unwanted_block_alert_get_piece_index(lt::unwanted_block_alert* alert) {
    return PieceIndex { .inner = static_cast<int32_t>(alert->piece_index) };
}

} // namespace ltrs

File: src/ffi/alerts/unwanted_block.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type unwanted_block_alert = crate::ffi::ffi::unwanted_block_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/unwanted_block.h");
        type PieceIndex = crate::ffi::ffi::PieceIndex;
        unsafe fn unwanted_block_alert_get_block_index(alert: *mut unwanted_block_alert) -> i32;
        unsafe fn unwanted_block_alert_get_piece_index(
            alert: *mut unwanted_block_alert,
        ) -> PieceIndex;
    }
}
